go-src/src/katharanp.go

```diff
@@ -113,14 +113,16 @@ func (k *KatharaNetworkPlugin) CreateEndpoint(req *network.CreateEndpointRequest
 	k.Lock()
 	defer k.Unlock()
 
-    /* Throw error if not in map */
+	/* Throw error if not in map */
 	if _, ok := k.networks[req.NetworkID]; !ok {
 		return nil, types.ForbiddenErrorf("%s network does not exist", req.NetworkID)
 	}
 
 	intfInfo := new(network.EndpointInterface)
 
-	if req.Interface == nil {
+	if req.Options["org.kathara.katharanp.mac"] != nil {
```

Comment:
Change the `org.kathara.katharanp.mac` option name to something simpler but meaningful.
I suggest `kathara.mac_addr`.